crates/sparrow-compiler/src/types/inference.rs

```diff
@@ -25,8 +25,16 @@ pub(crate) fn instantiate(
         "Arguments being instantiated should have been resolved against the signature"
     );
 
-    let mut types_for_constraints: HashMap<TypeConstraint, Vec<&Located<FenlType>>> =
-        HashMap::new();
+    let mut constraints_for_type_var: HashMap<TypeVariable, Vec<TypeConstraint>> = HashMap::new();
```

Comment:
I still don't understand why we need to collect this as a hash map. Why not just wait until we have the types for each variable, and then later iterate over the `type_parameters` themselves. As far as I can tell, this loop just creates a copy of the type parameters keyed by variable (duplicating the assertions you have elsewhere).

Reply:
You're correct - I missed removing this. This is unused.